File: frontend/src/api/designService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/designs';

export const designService = {
    saveDesign: async (userId, designData) => {
        const response = await axios.post(`${API_URL}/user/${userId}`, designData);
        return response.data;
    },

    getUserDesigns: async (userId) => {
        const response = await axios.get(`${API_URL}/user/${userId}`);
        return response.data;
    },

    getDesignById: async (id) => {
        const response = await axios.get(`${API_URL}/${id}`);
        return response.data;
    },

    deleteDesign: async (id) => {
        await axios.delete(`${API_URL}/${id}`);
    }
};

File: backend/src/main/java/com/printhub/service/DesignService.java
package com.printhub.service;

import com.printhub.model.SavedDesign;
import com.printhub.model.User;
import com.printhub.repository.SavedDesignRepository;
import com.printhub.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class DesignService {

    @Autowired
    private SavedDesignRepository designRepository;

    @Autowired
    private UserRepository userRepository;

    public SavedDesign saveDesign(Long userId, SavedDesign design) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("User not found"));
        design.setUser(user);
        return designRepository.save(design);
    }

    public List<SavedDesign> getUserDesigns(Long userId) {
        return designRepository.findByUserIdOrderByCreatedAtDesc(userId);
    }

    public Optional<SavedDesign> getDesignById(Long id) {
        return designRepository.findById(id);
    }

    public void deleteDesign(Long id) {
        designRepository.deleteById(id);
    }
}
